### PR TITLE
chore: bump to angular v17.1

### PR DESCRIPTION
**Pull request by Nicoss54**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Ng-zorro still use Angular 17.0

Issue Number: N/A

## What is the new behavior?

Ng Zorro use Angular 17.1 ansd is now up to date with the last version of Angular

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by Nicoss54** [comment]
> I think we should put the `yarn.lock` file into version control. We really need the lock file to lock dependency version information. WDYT? @Nicoss54 @OriginRing

Yeah for me it can be a good idea, in our case and to not change all the pipeline we can add the package-lock.json, and keep npm as our node package manager. I already tried to do that in the past, but @simplejason  refuse the change but I did'nt remember the reason why. 

@simplejason could you refresh my old memory  :)

**Comment by simplejason** [comment]
Since the version should be 17.2.0, I'll change some codes of this PR

**Comment by HyperLife1119** [comment]
> > I think we should put the `yarn.lock` file into version control. We really need the lock file to lock dependency version information. WDYT? @Nicoss54 @OriginRing
> 
> Yeah for me it can be a good idea, in our case and to not change all the pipeline we can add the package-lock.json, and keep npm as our node package manager. I already tried to do that in the past, but @simplejason refuse the change but I did'nt remember the reason why.
> 
> @simplejason could you refresh my old memory :) If all of you are agree to ahev a lock file, i will create a new PR which add the lock to the version control and also change a little bit the pipeline to use npm ci instead npm i

I'll be working on this after this PR has been merged and we've settled on using yarn instead of npm, consistent with angular/component.

**Comment by Nicoss54** [comment]
> I realized that this PR was not created according to the release guide and I need to recreate a PR. Since it has error during `:run:prerender`, so we must fix it and try again, otherwise the official website will not be deployed correctly :(

@simplejason, do have i to do somthing ? rename the PR ?

**Comment by Nicoss54** [comment]
@HyperLife1119 do you know  which change I have to made ?

File: components/tree-view/toggle.ts [hunk 1]
```diff
@@ -29,15 +27,7 @@ export class NzTreeNodeNoopToggleDirective {}
...
-  static ngAcceptInputType_recursive: BooleanInput;
-
-  @Input('nzTreeNodeToggleRecursive')
-  override get recursive(): boolean {
-    return this._recursive;
-  }
-  override set recursive(value: boolean) {
-    this._recursive = coerceBooleanProperty(value);
-  }
+  @Input({ alias: 'nzTreeNodeToggleRecursive', transform: coerceBooleanProperty }) override recursive = false;
```
[the remark this answers is not in the document]
Comment (Nicoss54) [reply components/tree-view/toggle.ts]: done

File: components/tree-view/padding.ts [hunk 2]
```diff
@@ -20,7 +20,9 @@ export class NzTreeNodePaddingDirective<T> extends CdkTreeNodePadding<T> {
     return this._level;
   }
   override set level(value: NumberInput) {
-    this._setLevelInput(value);
+    if (typeof value === 'number') {
+      this._setLevelInput(value);
+    }
```
[the remark this answers is not in the document]
Comment (Nicoss54) [reply components/tree-view/padding.ts]: done

File: package.json [hunk 3]
```diff
@@ -73,6 +73,7 @@
     "@typescript-eslint/eslint-plugin": "^6.10.0",
     "@typescript-eslint/parser": "^6.10.0",
     "antd-theme-generator": "1.2.2",
+    "browser-sync": "^3.0.0",
```
Comment (HyperLife1119) [inline package.json:76]: Can we remove this?

Comment (Nicoss54) [reply package.json:76]: This dependencies is added when we use ng update @angular/core @angular/cli

Comment (Nicoss54) [reply package.json:76]: This dependencies was added by the schematics update for @angular/ssr package

**Review by HyperLife1119 — APPROVED** [review]
LGTM

File: package.json (hunk 3 above)
Comment (Nicoss54) [reply package.json:76]: yes this package was added during the migration. but it seems there is no effect to remove it, i will push the package json file.

Saw you partially approved the PR, do i have to change something more ?

Comment (HyperLife1119) [reply package.json]: Yes, let's remove this dependency if we can.  You can push a change again.

Comment (Nicoss54) [reply package.json]: done :)

**Comment by Nicoss54** [comment]
@simplejason can we merge it ?

**Comment by Nicoss54** [comment]
@HyperLife1119 are you agree to merge this branch, seems nothing is broken in the pipeline and the preview of the doc is working correctly ? The you can work on the pipeline and yarn as package manager

**Review by HyperLife1119 — APPROVED** [review]
LGTM

**Pull request merged**